automation_script.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+
+# Move this file to the folder containing terraform.tfvars!
+# Also ensure the argo workflow is in the correct path as below
+
+# The following variables can be customised
+# some of them, like the project id, are required
+
+# The following contains paths for the terraform config files,
+# workflow file and namespace to use
+TERRAFORM_DIR="standard-gke-cluster-gcs"
+WORKFLOW_FILE="argo/argo_bucket_run.yaml"
+NAMESPACE="argo"
+
+# Cluster variables
+# Cluster name should be unique for better cost monitoring
+# In this case it is kept unique via including the timestamp
+PROJECT_ID=""
+REGION="europe-north1-b"
+TIMESTAMP=$(date +'%y%m%d-%H-%M')
+CLUSTER_NAME="cluster-$TIMESTAMP"
+NUM_NODES="3"
+MACHINE_TYPE="e2-standard-4"
+NODE_DISK_TYPE="pd-ssd"
+
+# Dataset variables
+RECID="30544"
+NUM_EVENTS="30000"
+NUM_JOBS="12"
+
+# Set a value for nfs disk type if using the nfs cluster, e.g. "pd-standard" or "pd-ssd"
+# If using the gcs (google cloud storage) bucket workflow, enter the name of your bucket
+NFS_DISK_TYPE=""
+BUCKET_NAME=""
+SERVICE_ACC_FILE=""
+
+# From this point, the actual script starts, first filling in variables from above into terraform.tfvars
+# and the argo workflow, to customise the number of jobs and how many events should be processed
+
+# Change the directory to the one containing the terraform configuration files
+cd "${TERRAFORM_DIR}"
+
+# Insert the variable values into the placeholders in terraform.tfvars
+sed -i.bak -e "s/<PROJECT_ID>/$PROJECT_ID/" -e "s/<REGION>/$REGION/" -e "s/<NAME>/$TIMESTAMP/" \
+    -e "s/<NUM_NODES>/$NUM_NODES/" -e "s/<MACHINE_TYPE>/$MACHINE_TYPE/" \
+    -e "s/<NODE_DISK_TYPE>/$NODE_DISK_TYPE/" -e "s/<NFS_DISK_TYPE>/$NFS_DISK_TYPE/" \
+    -e "s/<SERVICE_ACCOUNT_FILE>/$SERVICE_ACC_FILE/" "terraform.tfvars"
+
+# Insert the variable values into the argo workflow
+sed -i.bak -e "s/<NAME>/$TIMESTAMP/" -e "s/<RECID>/$RECID/" \
+    -e "s/<N_EVENTS>/$NUM_EVENTS/" -e "s/<N_JOBS>/$NUM_JOBS/" \
+    -e "s/<BUCKET_NAME>/$BUCKET_NAME/" "${WORKFLOW_FILE}"
```

Comment:
The configurable number of jobs could be changed here for the start (if added here)